servant/src/Servant/Utils/Enter.hs

```diff
@@ -39,6 +39,9 @@ instance ( Enter typ1 arg1 ret1, Enter typ2 arg2 ret2
 instance (Enter b arg ret) => Enter (a -> b) arg (a -> ret) where
     enter arg f a = enter arg (f a)
 
+instance Enter (arg -> ret) arg ret where
+    enter arg f = f arg
```

Comment:
I don't have strong opinion about this. Just that, we thought we were (almost) re-inventing `Enter` with our `HasArgument` class, but I get your point of doing more than one thing without sensible laws would be vague. I am all in for replicating `Enter`like class for `staticClient` purpose. WDYT @soenkehahn ?